### PR TITLE
Use GoogleTest test harness

### PR DESCRIPTION
**Pull request by ericniebler**
* add cmake magic to download and build gtest
* port any_unique_test to use gtest
* refs #39

### CODE REVIEW

**Comment by ericniebler** [comment]
Obviously, all the other tests need to be ported also to GTest.

**Comment by ericniebler** [comment]
We should separate examples from tests.

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ foreach(file-path ${test-sources})
     string( REPLACE ".cpp" "" file-path-without-ext ${file-path} )
     get_filename_component(file-name ${file-path-without-ext} NAME)
     add_executable( ${file-name} ${file-path})
-    target_link_libraries( ${file-name} PUBLIC unifex)
+    target_link_libraries(${file-name} PUBLIC unifex gtest_main)
```
Comment (lewissbaker) [inline examples/CMakeLists.txt]: I assume linking in gtest_main treat the `main()` it provides as a weak symbol and still prefer the `main()` from the sources?

Comment (ericniebler) [reply examples/CMakeLists.txt]: I _think_ so.

**Pull request merged**